### ISSUE TITLE
[Bug]: SM_120 error on RTX 5000 series

### ISSUE THREAD

**Issue by leoinpython** [IAHispano/Applio#969]
### Project Version

3.2.8

### Platform and OS Version

Windows 11 64 bit

### Affected Devices

All

### What happened?

not able to start the application

### Steps to reproduce

1. run-applio.bat
2. 
3. 
...

### Expected behavior

not able to output any files

### Attachments

![Image](https://github.com/user-attachments/assets/eb0fb104-f825-494e-8713-7e1bd56d9c5b)

### Screenshots or Videos

![Image](https://github.com/user-attachments/assets/f763def2-5f1a-4e06-bc34-c71848db7e05)

### Additional Information

installed newer pytorch via anaconda, 

![Image](https://github.com/user-attachments/assets/8f3d4292-fe24-40ef-b991-66f3b319811d)

**Comment by AznamirWoW** [comment]
You need to manually update torch to the version that supports 5000 series
but we have not tested that yet

**Comment by leoinpython** [comment]
> You need to manually update torch to the version that supports 5000 series but we have not tested that yet

I did update the Torch with anaconda, this last screenshot is the error after the update.

**Comment by AznamirWoW** [comment]
use a compiled version from huggingface
use `env\python -m pip install ...` to install torch/torchaudio/torchvision that support 5000 serties

it is likely you'll also need to get the latest code from the repository because torch 2.5.x no longer allows the load of weights without specifying weights_only flag.

**Comment by leoinpython** [comment]
> use a compiled version from huggingface use `env\python -m pip install ...` to install torch/torchaudio/torchvision that support 5000 serties
> 
> it is likely you'll also need to get the latest code from the repository because torch 2.5.x no longer allows the load of weights without specifying weights_only flag.

ya the purple screenshot is what i got after installing newer torch
I followed the instruction on torch
pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu126
im on Torch 2.6.0

**Comment by AznamirWoW** [comment]
numpy should be <2.0

**Comment by AznamirWoW** [comment]
from what I've gathered, you need to update torch modules to 2.6.0 and use Cuda Toolkit 12.8

torch wheel for windows is here
https://huggingface.co/w-e-w/torch-2.6.0-cu128.nv

**Comment by mbljoker** [comment]
@leoinpython 
You can replace the link https://download.pytorch.org/whl/cu121 with https://download.pytorch.org/whl/nightly/cu128.
Re-run the run-install.bat file.
The reason is that CUDA version 12.8 only started supporting the sm_120 architecture.

**Comment by leoinpython** [comment]
> [@leoinpython](https://github.com/leoinpython) You can replace the link https://download.pytorch.org/whl/cu121 with https://download.pytorch.org/whl/nightly/cu128. Re-run the run-install.bat file. The reason is that CUDA version 12.8 only started supporting the sm_120 architecture.

just did that, i think the requirement needs to be updated because when i run the install again, it'll just go to the old version again, and the same error pops up again, it just doesnt work.

**Comment by mbljoker** [comment]
@leoinpython 
Try replacing the whole line
pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --upgrade --index-url https://download.pytorch.org/whl/cu121 || goto :error
with
pip install torch torchvision torchaudio --upgrade --index-url https://download.pytorch.org/whl/nightly/cu128 || goto :error

**Comment by leoinpython** [comment]
> pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --upgrade --index-url https://download.pytorch.org/whl/cu121 || goto :error
> with
> pip install torch torchvision torchaudio --upgrade --index-url https://download.pytorch.org/whl/nightly/cu128 || goto :error

ya it did install the new version, but the startup of applio still show the same error and not usable

**Comment by AznamirWoW** [comment]
from Applio folder stand a command line prompt (cmd.exe)
then
`env\python -m pip uninstall torch torchaudio torchvision`
`env\python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu128`

**Comment by AznamirWoW** [comment]
> got these error instead.


just warinings

**Comment by leoinpython** [comment]
> > got these error instead.
> 
> just warinings

nah, so after i pressed any key to continue, it just disappear, doesnt even run anything.
previously if got the error of SM120, the interface would still load, just the generate windows will always shows error.

**Comment by AznamirWoW** [comment]
> > > got these error instead.
> > 
> > 
> > just warinings
> 
> nah, so after i pressed any key to continue, it just disappear, doesnt even run anything. previously if got the error of SM120, the interface would still load, just the generate windows will always shows error.

for libf0 issue please do
`env\python -m pip install libf0`

**Comment by leoinpython** [comment]
> > > > got these error instead.
> > > 
> > > 
> > > just warinings
> > 
> > 
> > nah, so after i pressed any key to continue, it just disappear, doesnt even run anything. previously if got the error of SM120, the interface would still load, just the generate windows will always shows error.
> 
> for libf0 issue please do `env\python -m pip install libf0`

this resolved everything now! thanks a lot

**Comment by alvar036** [comment]
above works, when training i do get these warnings/errors:

` Starting training...
Loaded pretrained (G) 'rvc/models/pretraineds/pretrained_v2/f0G48k.pth'
Loaded pretrained (D) 'rvc/models/pretraineds/pretrained_v2/f0D48k.pth'
C:\AI\APPLIO\ApplioV3.2.8-bugfix\rvc\train\train.py:440: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = GradScaler(enabled=config.train.fp16_run and device.type == "cuda")
  0%|                                                                                           | 0/84 [00:00<?, ?it/s]C:\AI\APPLIO\ApplioV3.2.8-bugfix\rvc\train\train.py:593: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast(enabled=use_amp):
C:\AI\APPLIO\ApplioV3.2.8-bugfix\rvc\train\train.py:608: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast(enabled=False):
C:\AI\APPLIO\ApplioV3.2.8-bugfix\rvc\train\train.py:620: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast(enabled=use_amp):
C:\AI\APPLIO\ApplioV3.2.8-bugfix\rvc\train\train.py:622: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast(enabled=False):
C:\AI\APPLIO\ApplioV3.2.8-bugfix\rvc\train\train.py:667: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with autocast(enabled=False):
JustinBieberV3 | epoch=1 | step=84 | time=22:20:18 | training_speed=0:00:39 `

Not sure if it actually is a problem since training started but posting it anyways!

**Comment by AznamirWoW** [comment]
you need to use the latest main branch code, old 3.2.8 does not support torch you've installed

**Comment by alvar036** [comment]
> you need to use the latest main branch code, old 3.2.8 does not support torch you've installed

I only see 3.2.8-bugfix as the latest release? or am i looking in the wrong place?

**Comment by AznamirWoW** [comment]
> > you need to use the latest main branch code, old 3.2.8 does not support torch you've installed
> 
> I only see 3.2.8-bugfix as the latest release? or am i looking in the wrong place?

https://github.com/IAHispano/Applio/archive/refs/heads/main.zip

**Comment by alvar036** [comment]
> > > you need to use the latest main branch code, old 3.2.8 does not support torch you've installed
> > 
> > 
> > I only see 3.2.8-bugfix as the latest release? or am i looking in the wrong place?
> 
> https://github.com/IAHispano/Applio/archive/refs/heads/main.zip

Thank you! works :D

**Comment by alvar036** [comment]
After training it gives me this when creating the index file, something seems wrong but not sure so here it is:

` Saved model 'C:\AI\APPLIO\Applio-main\logs\AA\G_2333333.pth' (epoch 200)
Saved model 'C:\AI\APPLIO\Applio-main\logs\AA\D_2333333.pth' (epoch 200)
Training has been successfully completed with 200 epoch, 16800 steps and 43.254 loss gen.
Lowest generator loss: 26.332 at epoch 35, step 2918
C:\AI\APPLIO\Applio-main\env\lib\site-packages\joblib\externals\loky\backend\context.py:136: UserWarning: Could not find the number of physical cores for the following reason:
[WinError 2] The system cannot find the file specified
Returning the number of logical cores instead. You can silence this warning by setting LOKY_MAX_CPU_COUNT to the number of cores you want to use.
  warnings.warn(
  File "C:\AI\APPLIO\Applio-main\env\lib\site-packages\joblib\externals\loky\backend\context.py", line 257, in _count_physical_cores
    cpu_info = subprocess.run(
  File "C:\AI\APPLIO\Applio-main\env\lib\subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\AI\APPLIO\Applio-main\env\lib\subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\AI\APPLIO\Applio-main\env\lib\subprocess.py", line 1456, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
Init 1/3 with method random
OpenBLAS warning: precompiled NUM_THREADS exceeded, adding auxiliary array for thread metadata.
Inertia for init 1/3: 863932.25
Init 2/3 with method random
Inertia for init 2/3: 857140.4375
Init 3/3 with method random
Inertia for init 3/3: 861618.0
[MiniBatchKMeans] Reassigning 4096 cluster centers.
Minibatch step 1/3912: mean batch inertia: 34.63676452636719
[MiniBatchKMeans] Reassigning 3717 cluster centers.
Minibatch step 2/3912: mean batch inertia: 33.786720275878906, ewa inertia: 33.786720275878906
Minibatch step 3/3912: mean batch inertia: 30.210317611694336, ewa inertia: 33.60391751039804 `

**Comment by AznamirWoW** [comment]
latest nightly messes up the training

`env\python -m pip install --pre torch==2.7.0.dev20250311 torchvision torchaudio==2.6.0.dev20250312 --index-url https://download.pytorch.org/whl/nightly/cu128`

**Comment by AznamirWoW** [comment]
> After training it gives me this when creating the index file, something seems wrong but not sure so here it is:
> 
> `Saved model 'C:\AI\APPLIO\Applio-main\logs\AA\G_2333333.pth' (epoch 200) Saved model 'C:\AI\APPLIO\Applio-main\logs\AA\D_2333333.pth' (epoch 200) Training has been successfully completed with 200 epoch, 16800 steps and 43.254 loss gen. Lowest generator loss: 26.332 at epoch 35, step 2918 C:\AI\APPLIO\Applio-main\env\lib\site-packages\joblib\externals\loky\backend\context.py:136: UserWarning: Could not find the number of physical cores for the following reason: [WinError 2] The system cannot find the file specified Returning the number of logical cores instead. You can silence this warning by setting LOKY_MAX_CPU_COUNT to the number of cores you want to use. warnings.warn( File "C:\AI\APPLIO\Applio-main\env\lib\site-packages\joblib\externals\loky\backend\context.py", line 257, in _count_physical_cores cpu_info = subprocess.run( File "C:\AI\APPLIO\Applio-main\env\lib\subprocess.py", line 503, in run with Popen(*popenargs, **kwargs) as process: File "C:\AI\APPLIO\Applio-main\env\lib\subprocess.py", line 971, in __init__ self._execute_child(args, executable, preexec_fn, close_fds, File "C:\AI\APPLIO\Applio-main\env\lib\subprocess.py", line 1456, in _execute_child hp, ht, pid, tid = _winapi.CreateProcess(executable, args, Init 1/3 with method random OpenBLAS warning: precompiled NUM_THREADS exceeded, adding auxiliary array for thread metadata. Inertia for init 1/3: 863932.25 Init 2/3 with method random Inertia for init 2/3: 857140.4375 Init 3/3 with method random Inertia for init 3/3: 861618.0 [MiniBatchKMeans] Reassigning 4096 cluster centers. Minibatch step 1/3912: mean batch inertia: 34.63676452636719 [MiniBatchKMeans] Reassigning 3717 cluster centers. Minibatch step 2/3912: mean batch inertia: 33.786720275878906, ewa inertia: 33.786720275878906 Minibatch step 3/3912: mean batch inertia: 30.210317611694336, ewa inertia: 33.60391751039804`

this does not seem 5000-series related.

try opening a command prompt from applio folder and running ``env\python core.py index --model_name your_model_name`` and see if that happens
